modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1188,25 +1188,16 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
         debug('Input details: %O', signatureContext);
 
         const sigHashType = utxolib.bitgo.getDefaultSigHash(self.network);
+        const redeemScript = signatureContext.unspent.redeemScript
+          ? Buffer.from(signatureContext.unspent.redeemScript, 'hex')
+          : undefined;
+        const witnessScript = signatureContext.unspent.witnessScript
+          ? Buffer.from(signatureContext.unspent.witnessScript, 'hex')
+          : undefined;
+        const scriptType = witnessScript ? (redeemScript ? 'p2shP2wsh' : 'p2wsh') : 'p2sh';
+        debug(`Signing ${scriptType} input`);
```

Comment:
I decided to move the hex decoding outside the `try` loop because errors occurring there are not really _signing_ errors